cfme/tests/ansible/test_embedded_ansible_basic.py

```diff
@@ -317,20 +288,28 @@ def test_control_add_ansible_playbook_action_invalid_address(request, catalog_it
         fauxfactory.gen_alphanumeric(),
         action_type="Run Ansible Playbook",
         action_values={
-            "run_ansible_playbook":
-            {
+            "run_ansible_playbook": {
                 "playbook_catalog_item": catalog_item.name,
                 "inventory": {
                     "specific_hosts": True,
-                    "hosts": "invalid_address_!@#$%^&*"
-                }
+                    "hosts": "invalid_address_!@#$%^&*",
+                },
             }
-        }
+        },
     )
-    request.addfinalizer(action.delete_if_exists)
+
+    @request.addfinalizer
+    def _finalizer():
+        if action.exists:
+            action_id = appliance.rest_api.collections.actions.get(
+                description=action.description
+            ).id
+            appliance.rest_api.collections.actions.action.delete(id=action_id)
```

Comment:
**Required** This finalizer seems to be used twice here, perhaps break it out into a separate method.

Reply:
@jawatts Updating it.